test/org/joml/test/RectangleiTest.java

```diff
@@ -74,6 +74,73 @@ public void testRectangleContains() {
         Assert.assertEquals(first.intersection(second, new Rectanglei()), new Rectanglei(0, 0, 1, 1));
     }
 
+    public void testRectangleEdgeIntersection() {
+        Rectanglei center = new Rectanglei(0, 0, 1, 1);
+        Rectanglei right = new Rectanglei(1, 0, 2, 1);
+        Rectanglei left = new Rectanglei(-1, 0, 0, 1);
+        Rectanglei top = new Rectanglei(0, 1, 1, 2);
+        Rectanglei bottom = new Rectanglei(0, -1, 1, 0);
+
+        Assert.assertTrue(center.isValid());
+        Assert.assertTrue(right.isValid());
+        Assert.assertTrue(left.isValid());
+        Assert.assertTrue(top.isValid());
+        Assert.assertTrue(bottom.isValid());
+
+        Assert.assertFalse(center.containsRectangle(right));
+        Assert.assertFalse(center.containsRectangle(left));
+        Assert.assertFalse(center.containsRectangle(top));
+        Assert.assertFalse(center.containsRectangle(bottom));
+
+        Assert.assertTrue(center.intersectsRectangle(right));
+        Assert.assertFalse(center.intersectsRectangle(left));
+        Assert.assertTrue(center.intersectsRectangle(top));
+        Assert.assertFalse(center.intersectsRectangle(bottom));
+
+        Assert.assertFalse(right.intersectsRectangle(center));
+        Assert.assertTrue(left.intersectsRectangle(center));
+        Assert.assertFalse(top.intersectsRectangle(center));
+        Assert.assertTrue(bottom.intersectsRectangle(center));
+    }
```

Comment:
Same comment as in https://github.com/JOML-CI/JOML/pull/247#discussion_r515649255 applies. Let's try to write specific test cases to be able to pin-point problems to the actual root cause faster (or to look up specific properties more easily).